charts/revere/templates/networkPolicy.yaml

```diff
@@ -0,0 +1,69 @@
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: {{ .Release.name }}-networkpolicy
+  labels:
+    {{- include "revere.labels" . | nindent 4 }}
+spec:
+  policyTypes:
+  {{- if .Values.networkPolicy.ingress.enabled }}
+  - Ingress
+  {{- end }}
+  {{- if .Values.networkPolicy.egress.enabled }}
```

Comment:
Does App sec care about both ingress and egress here?

Reply:
Egress is how we can prevent Revere from talking to ArgoCD, so that's the main one they want, even though egress is much harder than ingress to maintain. As for ingress, it is a way we could let only VPNed Broadies access Revere. I don't know if we'd implement that here or with a more classical app armor rule.

Reply:
This was a poorly phrased question by me. I guess what I'm really asking is can these egress configs prevent intra cluster communications? Ie over kubernetes internal DNS? Also if this question doesn't make sense that is totally fine, we can chat about it next week.

Reply:
Maybe this might be a question to ask Denis. The k8s docs and every tutorial out there talk about different mechanisms for allowing pods to talk to each other through network policies, implying that it does indeed block in-cluster traffic, but it is a good point that we actually care about the opposite, we want a bulletproof firewall